src/app/modules/shared-components/components/access-code-view/access-code-view.component.ts

```diff
@@ -31,11 +31,15 @@ export class AccessCodeViewComponent {
     this.joinByCodeService.checkCodeValidity(this.code).subscribe(response => {
       this.state = 'ready';
 
-      if (!response.valid) {
+      if (!response.valid && response.reason) {
```

Comment:
Why is this added condition `response.reason` needed ?

Reply:
By doc reason is optional https://france-ioi.github.io/algorea-devdoc/api/#operation/groupsCodeCheck

Reply:
I think it should be handled at the same place, that way we can clearly see at first glance that no matter what, when the response is invalid, we display an error message, ie:
```ts
if (!response.valid) {
  this.actionFeedbackService.error(response.reason ? … : fallback value)
}
```

For the error message itself, since it's displayed to the end user, it should not contain tech-related information.
Here, we check the code validity, so if the response is invalid, I guess it means that the code is invalid. The message could be something like `"The provided code is invalid"` ?

Reply:
Fixed